zilliqa/tests/it/persistence.rs

```diff
@@ -235,11 +244,28 @@ async fn checkpoints_test(mut network: Network) {
         ..Default::default()
     });
 
+    // Confirm wallet and new_node_wallet have the same block and state
     let new_node_wallet = network.wallet_of_node(new_node_idx).await;
-    let latest_block = new_node_wallet.get_block_number().await.unwrap();
-    assert_eq!(latest_block, 10.into());
+    let latest_block_number = new_node_wallet.get_block_number().await.unwrap();
+    assert_eq!(latest_block_number, 10.into());
+
+    let block = wallet
+        .get_block(latest_block_number)
+        .await
+        .unwrap()
+        .unwrap();
+    assert_eq!(block.transactions.len(), 1);
+
+    let block_from_checkpoint = new_node_wallet
+        .get_block(latest_block_number)
+        .await
+        .unwrap()
+        .unwrap();
+    assert_eq!(block.transactions, block_from_checkpoint.transactions);
+    // Check access to previous block state via fetching author of current block
```

Comment:
Just double checking that this test fails in `main`?

Reply:
yes, the line

```rust
assert_eq!(block.author, block_from_checkpoint.author);
```
was the test harness as it fails for when state of parent doesnt exist in trie